### PR TITLE
Make `repo-age` a clickable link

### PR DESCRIPTION
**Pull request by yakov116**
Follows 
>Hardik can you implement the link to the first commit as well? https://github.com/sindresorhus/refined-github/pull/2667/files#diff-9d82f672e2199b6b2b455fa80d3373f2R86

>_Originally posted by @fregante in https://github.com/sindresorhus/refined-github/pull/2783#issuecomment-636680756_

This will cause issues, with the cache due to the href being empty. I am not sure how to go about it.

### CODE REVIEW

File: source/features/repo-age.tsx [hunk 1]
```diff
@@ -30,21 +30,22 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
 	if (commitsCount === 1) {
-		return select('.commit-tease-sha + span relative-time')!.attributes.datetime.value;
+		return [select('.commit-tease-sha + span relative-time')!.attributes.datetime.value, commitUrl];
 	}
 
-	const relativeTime = await fetchDom(
+	const commit = await fetchDom(
 		`${getRepoURL()}/commits?after=${commitSha}+${commitsCount - 2}`,
-		'.commit-meta relative-time'
+		'.commit-group .commit'
```
Comment (fregante) [inline source/features/repo-age.tsx:38]: Selectors don't have to be super specific unless it's necessary
```suggestion
		'.commit'
```

File: source/features/repo-age.tsx [hunk 2]
```diff
@@ -30,21 +30,22 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
 	);
-
-	return relativeTime!.attributes.datetime.value;
+	const timeStamp = select('.commit-meta relative-time', commit)!.attributes.datetime.value;
```
Comment (fregante) [inline source/features/repo-age.tsx:40]:
```suggestion
	const timeStamp = select('relative-time', commit)!.attributes.datetime.value;
```

File: source/features/repo-age.tsx [hunk 3]
```diff
@@ -30,21 +30,22 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
+	const href = select<HTMLAnchorElement>('a.message', commit)!.href;
```
Comment (fregante) [inline source/features/repo-age.tsx:41]:
```suggestion
	const {href} = select<HTMLAnchorElement>('a.message', commit)!;
```

File: source/features/repo-age.tsx [hunk 4]
```diff
@@ -30,21 +30,22 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
+	return [timeStamp, href];
 }, {
 	cacheKey: () => __filebasename + ':' + getRepoURL()
 });
 
 async function init(): Promise<void> {
-	const firstCommitDate = await getFirstCommitDate();
+	const [firstCommitDate, firstCommitHref] = await getFirstCommitDate() ?? [];
```
Comment (fregante) [inline source/features/repo-age.tsx:48]: Update function names when you change their signature
```suggestion
	const [firstCommitDate, firstCommitHref] = await getFirstCommit() ?? [];
```

File: source/features/repo-age.tsx [hunk 5]
```diff
@@ -30,21 +30,22 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
-		'.commit-meta relative-time'
+		'.commit-group .commit'
 	);
-
-	return relativeTime!.attributes.datetime.value;
+	const timeStamp = select('.commit-meta relative-time', commit)!.attributes.datetime.value;
+	const href = select<HTMLAnchorElement>('a.message', commit)!.href;
+	return [timeStamp, href];
 }, {
 	cacheKey: () => __filebasename + ':' + getRepoURL()
```
Comment (fregante) [inline source/features/repo-age.tsx:44]: This solves the "old cache" issue
```suggestion
	cacheKey: () => __filebasename + ':' + getRepoURL(),
	isExpired: value => typeof value === 'string'
```

File: source/features/repo-age.tsx (hunk 3 above)
Comment (yakov116) [reply source/features/repo-age.tsx:41]: Debated this one 👍

File: source/features/repo-age.tsx [hunk 6]
```diff
@@ -30,21 +30,23 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
 	);
-
-	return relativeTime!.attributes.datetime.value;
+	const timeStamp = select('relative-time', commit)!.attributes.datetime.value;
+	const {href} = select<HTMLAnchorElement>('a.message', commit)!;
+	return [timeStamp, href];
 }, {
-	cacheKey: () => __filebasename + ':' + getRepoURL()
+	cacheKey: () => __filebasename + ':' + getRepoURL(),
+	isExpired: value => typeof value === 'string'
```
Comment (yakov116) [inline source/features/repo-age.tsx:45]: This line needs typescript help

File: source/features/repo-age.tsx [hunk 7]
```diff
@@ -16,7 +16,7 @@ const dateFormatter = new Intl.DateTimeFormat('en-US', {
 	day: 'numeric'
 });
 
-const getFirstCommitDate = cache.function(async (): Promise<string | undefined> => {
+const getFirstCommit = cache.function(async (): Promise<string[] | undefined> => {
```
Comment (fregante) [inline source/features/repo-age.tsx:19]:
```suggestion
const getFirstCommit = cache.function(async (): Promise<[string, string] | undefined> => {
```

**Comment by yakov116** [comment]
Give me one min let me test it

File: source/features/repo-age.tsx [hunk 8]
```diff
@@ -30,21 +30,23 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
+	const commit = await fetchDom(
 		`${getRepoURL()}/commits?after=${commitSha}+${commitsCount - 2}`,
-		'.commit-meta relative-time'
+		'.commit'
 	);
-
-	return relativeTime!.attributes.datetime.value;
+	const timeStamp = select('relative-time', commit)!.attributes.datetime.value;
+	const {href} = select<HTMLAnchorElement>('a.message', commit)!;
+	return [timeStamp, href];
```
Comment (fregante) [inline source/features/repo-age.tsx:42]:
```suggestion
	const {pathname} = select<HTMLAnchorElement>('a.message', commit)!;
	return [timeStamp, pathname];
```

File: source/features/repo-age.tsx [hunk 9]
```diff
@@ -30,21 +30,23 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
 	);
-
-	return relativeTime!.attributes.datetime.value;
+	const timeStamp = select('relative-time', commit)!.attributes.datetime.value;
+	const {pathname} = select<HTMLAnchorElement>('a.message', commit)!;
+	return [timeStamp, pathname];
 }, {
-	cacheKey: () => __filebasename + ':' + getRepoURL()
+	cacheKey: () => __filebasename + ':' + getRepoURL(),
+	shouldRevalidate: value => typeof value === 'string'
```
Comment (yakov116) [inline source/features/repo-age.tsx:45]:
```suggestion
	shouldRevalidate: value => typeof value === 'string' // Rmove after June 2020
```
WDYT?

File: source/features/repo-age.tsx [hunk 10]
```diff
@@ -30,21 +30,23 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
 	);
-
-	return relativeTime!.attributes.datetime.value;
+	const timeStamp = select('relative-time', commit)!.attributes.datetime.value;
+	const {pathname} = select<HTMLAnchorElement>('a.message', commit)!;
+	return [timeStamp, pathname];
 }, {
-	cacheKey: () => __filebasename + ':' + getRepoURL()
+	cacheKey: () => __filebasename + ':' + getRepoURL(),
+	shouldRevalidate: value => typeof value === 'string' // Rmove after June 2020
```
Comment (fregante) [inline source/features/repo-age.tsx:45]:
```suggestion
	shouldRevalidate: value => typeof value === 'string' // TODO: Remove after June 2020
```

File: source/features/repo-age.tsx [hunk 11]
```diff
@@ -30,21 +30,23 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
+	const commit = await fetchDom(
 		`${getRepoURL()}/commits?after=${commitSha}+${commitsCount - 2}`,
-		'.commit-meta relative-time'
+		'.commit'
 	);
-
-	return relativeTime!.attributes.datetime.value;
+	const timeStamp = select('relative-time', commit)!.attributes.datetime.value;
+	const {pathname} = select<HTMLAnchorElement>('a.message', commit)!;
+	return [timeStamp, pathname];
```
Comment (fregante) [inline source/features/repo-age.tsx:42]: Kinda:
```suggestion
	return [new Date(timeStamp).getTime(), pathname];
```

File: source/features/repo-age.tsx [hunk 12]
```diff
@@ -30,21 +30,23 @@ const getFirstCommitDate = cache.function(async (): Promise<string | undefined>
...
 }, {
-	cacheKey: () => __filebasename + ':' + getRepoURL()
+	cacheKey: () => __filebasename + ':' + getRepoURL(),
+	shouldRevalidate: value => typeof value === 'string' // TODO: Remove after June 2020
 });
 
 async function init(): Promise<void> {
-	const firstCommitDate = await getFirstCommitDate();
+	const [firstCommitDate, firstCommitHref] = await getFirstCommit() ?? [];
```
Comment (fregante) [inline source/features/repo-age.tsx:49]:
```suggestion
	const [firstCommitDate, firstCommitUrl] = await getFirstCommit() ?? [];
```

**Comment by yakov116** [comment]
cache cant store a date

File: source/features/repo-age.tsx (hunk 11 above)
Comment (yakov116) [reply source/features/repo-age.tsx]: We need the date for the title, so we cant just store the time

**Comment by fregante** [comment]
> We need the date for the title, so we cant just store the time

You were right on this one. Changing the code to store **both the text and the timestamp** isn't any better than just storing the text.

**Pull request merged**